### ISSUE TITLE
A11y Audit: [calcite-radio-group-item] - missing associated label describing its purpose

### ISSUE THREAD

**Issue by caripizza** [Esri/calcite-components#3431]
### Issue Summary

[Issue]
Radio Group Items do not provide a programmatically associated label that describes the purpose of the radio button. 

[User Impact]
Users may not be able to fill out form fields correctly if the visible label is not programmatically associated with its field.

[Code Reference]
`<calcite-radio-group-item value="react" tabindex="0" aria-checked="true" role="radio" icon-position="start" calcite-hydrated="" checked="">React</calcite-radio-group-item>`

### Related Standards
https://amp.levelaccess.net/public/reporting/view_pattern.php?pattern_id=175260

### Reproduction Steps

Add Radio Group Items to a page and notice they don't have the aria-label attribute, ex:

```html
    <calcite-radio-group layout="horizontal" appearance="solid" scale="m" width="auto">
      <calcite-radio-group-item value="apple" checked>Apple</calcite-radio-group-item>
      <calcite-radio-group-item value="mango">Mango</calcite-radio-group-item>
      <calcite-radio-group-item value="tomato">Tomato</calcite-radio-group-item>
      <calcite-radio-group-item value="banana">Banana</calcite-radio-group-item>
    </calcite-radio-group>
```

### Expected Behavior

Radio Group Items have an associated label that describes their purpose.

[Recommendation]
Developer should provide "aria-label" attribute to the element with role="radio".

[Compliant Code Example]
`<calcite-radio-group-item value="react" tabindex="0" aria-checked="true" aria-label="react" role="radio" icon-position="start" calcite-hydrated="" checked="">React</calcite-radio-group-item>`

### Remediation Steps

n/a

### Screenshots/Images

n/a

### Additional Context

n/a

### Issue Source

- Instance: https://amp.levelaccess.net/public/reporting/view_instance.php?instance_id=1901833855

**Comment by benelan** [comment]
verified on next
